### PR TITLE
teamspeak-server package & module maintenance

### PR DESCRIPTION
**Pull request by Shados**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
- First commit updates the `teamspeak-server` package and makes the `tsdnsserver` binary available in its output. I might later submit a NixOS module for TSDNS, as it looks like I'd have to use it to avoid a long-standing race condition in the client's DNS SRV resolution algorithm.
- Second commit is some housekeeping on the NixOS module, should be essentially transparent to any users.

### CODE REVIEW

File: nixos/modules/services/networking/teamspeak3.nix [hunk 1]
```diff
@@ -96,34 +95,34 @@ in
...
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    users.extraUsers.teamspeak =
-      { name = "teamspeak";
+  config = mkMerge [
+    (mkIf cfg.enable {
+      users.users.teamspeak = {
+        name = "teamspeak";
```
Comment (joachifm) [inline nixos/modules/services/networking/teamspeak3.nix]: If you're doing cleanup, the `name` is redundant here, it defaults to the `users` attr key.

File: nixos/modules/services/networking/teamspeak3.nix [hunk 2]
```diff
@@ -96,34 +95,34 @@ in
...
         group = group;
         uid = config.ids.uids.teamspeak;
+        home = cfg.dataDir;
+        createHome = true;
       };
 
-    users.extraGroups.teamspeak =
-      { name = "teamspeak";
+      users.groups.teamspeak = {
+        name = "teamspeak";
```
Comment (joachifm) [inline nixos/modules/services/networking/teamspeak3.nix]: `name` is redundant here as well.

File: nixos/modules/services/networking/teamspeak3.nix [hunk 3]
```diff
@@ -136,7 +135,9 @@ in
           PermissionsStartOnly = true; # preStart needs to run with root permissions
```
Comment (joachifm) [inline nixos/modules/services/networking/teamspeak3.nix]: The inline comment here is redundant

**Comment by joachifm** [comment]
LGTM, added a few notes you may want to consider if yoǘ're doing cleanup anyway.

**Comment by Shados** [comment]
@joachifm Figured I might as well while I'm at it, so I've amended the commit with that further cleanup :).

**Comment by joachifm** [comment]
Thank you

**Pull request merged**